client/src/modules/inventory/configuration/groups/groups.js

```diff
@@ -26,18 +27,24 @@ function InventoryGroupsController($translate, InventoryGroup, Account, Notify,
   // startup
   startup();
 
+  function handler(err) {
+    if (err) {
```

Comment:
Why not using `Notify` even if ```err``` is ```undefined``` and print a general error message?

Reply:
Because when `err` does not exist, it means the modal has been canceled by the user.  The way it works is like this:
 1. User clicks the [cancel button](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/modules/inventory/configuration/groups/modals/actions.tmpl.html#L72) which calls [cancel()](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/modules/inventory/configuration/groups/modals/actions.modal.js#L64) in the controller.
 2. This fires the [ModalInstance.dismiss()](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/modules/inventory/configuration/groups/modals/actions.modal.js#L65) method.  According to [UI Bootstrap](https://angular-ui.github.io/bootstrap/#!#modal) this `dismiss()` method rejects the promise returned by the modal.
 3. The rejection makes the promise chain terminate in [this .catch() statement](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/client/src/modules/inventory/configuration/groups/groups.js#L40).  However, since the `ModalInstance.dismiss()` method wasn't called with any parameters, the `err` parameter is empty.  This tells us that it wasn't an actual error from the HTTP request - those all throw real errors.  It was just a cancellation by the user.